app/controllers/auth/NinoAuthAction.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.auth
+
+import uk.gov.hmrc.auth.core.retrieve.v2.Retrievals
+import uk.gov.hmrc.auth.core.{AuthorisedFunctions, ConfidenceLevel, CredentialStrength, InsufficientConfidenceLevel, InsufficientEnrolments}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.bootstrap.auth.DefaultAuthConnector
+import javax.inject.Inject
+
+import scala.concurrent.{ExecutionContext, Future}
+
+sealed trait AtsNino
+case class SuccessAtsNino(nino: String) extends AtsNino
+object NoAtsNinoFound extends AtsNino
+object UpliftRequiredAtsNino extends AtsNino
+
+class NinoAuthAction @Inject()(override val authConnector: DefaultAuthConnector)(implicit ec: ExecutionContext)
+    extends AuthorisedFunctions {
+
+  def getNino()(implicit hc: HeaderCarrier): Future[AtsNino] =
+    authorised(ConfidenceLevel.L200 and CredentialStrength(CredentialStrength.strong)).retrieve(Retrievals.nino) {
```

Comment:
Missing a test for when the CredStrength is weak (we should just show them the not authorised page for now)